docs/core/entity/climate.md

```diff
@@ -139,6 +139,48 @@ class MyClimateEntity(ClimateEntity):
         """Set new target hvac mode."""
 ```
 
+### Turn on
+
+```python
+class MyClimateEntity(ClimateEntity):
+    # Implement one of these methods if the device supports it.
+    # Make sure that hvac_mode is properly adjusted as well.
```

Comment:
This comment, and the similar one for turn off is a bit unclear / misleading. The hvac_mode doesn't have to be optimistically set by the `turn_on` or `async_turn_on`, it should instead be set when we get a state update from the remote device in most cases.

Reply:
Yes, that is what I meant here. I just had trouble wording that clearly and yet succinctly so I wrote it in the most general terms I could find.
Maybe you have some proposals for better phrasing?